app/src/main/java/org/mozilla/focus/autocomplete/AutocompleteAddFragment.kt

```diff
@@ -60,10 +60,17 @@ class AutocompleteAddFragment : Fragment() {
                     .toLowerCase()
                     .removePrefixesIgnoreCase("http://", "https://", "www.")
 
-            if (domain.isEmpty()) {
-                domainView.error = getString(R.string.preference_autocomplete_add_error)
-            } else {
-                saveDomainAndClose(activity.applicationContext, domain)
+            launch(UI) {
+                when {
+                    domain.isEmpty()
+                    -> domainView.error = getString(R.string.preference_autocomplete_add_error)
+                    CustomDomains
```

Comment:
I don't think this is correct style. You want to keep the arrows on the same line like:
```kotlin
domain.isEmpty() -> {
    // Multiline things here
}
```

And I'd avoid having multi-line expressions as the switch expression in the when statement (e.g. store the result as a variable above). I think they should be pretty literal: "when this is an error", "when this is a success", etc.

Reply:
@mcomella good catch, would this be difficult to add to ktlint or detektcheck?

Reply:
I don't know how easy ktlint or detektCheck are to extend. detektCheck may have a check like this built in: I'm not sure how strict our configuration is.